packages/salesforcedx-vscode-core/src/conflict/directoryDiffer.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as fs from 'fs';
+import * as path from 'path';
+
+export interface DirectoryDiffResults {
+  different: Set<string>;
+  scannedLocal: number;
+  scannedRemote: number;
+}
+
+export interface DirectoryDiffer {
+  diff(localSourcePath: string, remoteSourcePath: string): DirectoryDiffResults;
+}
+
+interface FileStats {
+  filename: string;
+  subdir: string;
+  relPath: string;
+}
+
+export class CommonDirDirectoryDiffer implements DirectoryDiffer {
+  constructor() {}
+
+  public diff(
+    localSourcePath: string,
+    remoteSourcePath: string
+  ): DirectoryDiffResults {
+    const localSet = this.listFiles(localSourcePath);
+    const different = new Set<string>();
+
+    // process remote files to generate differences
+    let scannedRemote = 0;
+    this.walkFiles(remoteSourcePath, '', stats => {
+      scannedRemote++;
+      if (localSet.has(stats.relPath)) {
+        const file1 = path.join(localSourcePath, stats.relPath);
+        const file2 = path.join(remoteSourcePath, stats.relPath);
+        if (this.filesDiffer(file1, file2)) {
+          different.add(stats.relPath);
```

Comment:
Eventually we will need the full filepaths of the temp file and the local (workspace) file when there's conflicts so we can pipe them into the visual diff api on vscode. We might need to modify the results returned here but I don't think visual diff is part for the first phase, right ?

Reply:
The root paths are available where these results are checked, so we do have the full path available there. Visualization is not part of the first phase, we have a meeting next week to discuss what the visualization might look like.